portability-transfer/src/main/java/org/dataportabilityproject/transfer/PortabilityInMemoryDataCopier.java

```diff
@@ -121,16 +122,72 @@ private void copyHelper(
             exportAuthData,
             importAuthData,
             Optional.of(new ExportInformation(
-                continuationData.getPaginationData(), exportInformation.get().getContainerResource())));
+                continuationData.getPaginationData(),
+                exportInformation.get().getContainerResource())));
       }
 
       // Start processing sub-resources
       if (continuationData.getContainerResources() != null
           && !continuationData.getContainerResources().isEmpty()) {
         for (ContainerResource resource : continuationData.getContainerResources()) {
-          copyHelper(jobId, exportAuthData, importAuthData, Optional.of(new ExportInformation(null, resource)));
+          copyHelper(jobId, exportAuthData, importAuthData,
+              Optional.of(new ExportInformation(null, resource)));
+        }
+      }
+    }
+  }
+
+  private ExportResult runExportLogic(UUID jobId, AuthData exportAuthData,
+      Optional<ExportInformation> exportInformation) {
+    ExportResult<?> exportResult = exportHelper(jobId, exportAuthData, exportInformation);
+    if (exportResult.getType() == ResultType.ERROR) {
```

Comment:
is it possible to simplify this to consolidate the checking exportResult.getType multiple times and the manual increment of retries? something like:

resultType = get initial result . type;
for (int attempt = 0; resultType == error; attempt++) {
   resultType = retried result.type
}

Reply:
That's much cleaner, thank you!  Forgotten you could do that.